exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -374,17 +374,22 @@ public void run () {
     }
 
     private void pollShutdown(Drillbit drillbit) throws IOException, InterruptedException {
-      final Path drillPidDirPath = FileSystems.getDefault().getPath(System.getenv("DRILL_PID_DIR"));
-      final String gracefulFileName = System.getenv("GRACEFUL_SIGFILE");
+      final String drillHome = System.getenv("DRILL_HOME");
+      final String gracefulFile = System.getenv("GRACEFUL_SIGFILE");
+      if (drillHome == null || gracefulFile == null) {
+        logger.warn("Cannot access graceful file. Graceful shutdown from command line will not be supported.");
+        return;
+      }
+      final Path drillHomePath = Paths.get(drillHome);
```

Comment:
1. We might check if `drillHomePath` actually exists and reachable and issue a warning rather than fail if something is going wrong.
2. Please update PR and commit error description, it turned out NPE exists on other systems as well (just remove `on Windows part`).

Reply:
@arina-ielchiieva 
1. Added try-catch block while getting the drillHome path. What do you mean by if it is not set correctly? Both script and the code access the same variable and create or look for the graceful file.
2. Updated the PR.
3. Added java doc.